erpnext/public/js/utils/sales_common.js

```diff
@@ -369,6 +369,11 @@ erpnext.sales_common = {
 					}
 				}
 			}
+
+			coupon_code() {
```

Comment:
This will be triggered even when the coupon code is set.
Shouldn't this check if the coupon code is removed and then set the discount amounts as 0?

Reply:
When the coupon code is set the server side logic fetches the discount amounts when save is triggered to overwrite whatever is present in the discount amount currently according to the coupon code. Thought of skipping the condition because anyway it's fetched on save and if user sets eg: 14 % discount percentage and corresponding discount amount and then sets the coupon code for 12 %, these values are overridden by the server-side logic on save. So I thought emptying them when coupon code is changed makes more sense. If its better to put the condition in i can add it.